### PR TITLE
[dashboard] give team URLs a dedicated slug

### LINKED ISSUE #5465: [teams] move team URLs in its own slug (namespace)
To avoid name clashes we should introduce a `/team/` segment to the team URLs.
E.g. `gitpod.io/team/<team-name>`

### PR DESCRIPTION
**Pull request by svenefftinge**
fixes #5465 - [teams] move team URLs in its own slug (namespace)

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **svenefftinge** after the PR has been reviewed.

Associated issue: *#5465*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by AlexTugarev** [comment]
route matchers need to be adjusted as well. for instance in `ConfigureProject.tsx` we have `useRouteMatch<{ teamSlug: string, projectSlug: string }>("/:teamSlug/:projectSlug/configure")`.

File: components/dashboard/src/projects/ConfigureProject.tsx [hunk 1]
```diff
@@ -58,7 +58,7 @@ export default function () {
   const { teams } = useContext(TeamsContext);
   const location = useLocation();
   const team = getCurrentTeam(location, teams);
-  const routeMatch = useRouteMatch<{ teamSlug: string, projectSlug: string }>("/:teamSlug/:projectSlug/configure");
+  const routeMatch = useRouteMatch<{ teamSlug: string, projectSlug: string }>("/t/:teamSlug/:projectSlug/configure");
```
Comment (AlexTugarev) [inline components/dashboard/src/projects/ConfigureProject.tsx:61]: I'm afraid there are more instances of `useRouteMatch` to adjust, e.g. in Projects.tsx

Comment (svenefftinge) [reply components/dashboard/src/projects/ConfigureProject.tsx:61]: Ok, thanks. I have updated those. Any other leftovers you see? 
It's a little hard to find all the references ...

Comment (AlexTugarev) [reply components/dashboard/src/projects/ConfigureProject.tsx:61]: just searched for all `useRouteMatch` usages and it looks good now.

Comment (svenefftinge) [reply components/dashboard/src/projects/ConfigureProject.tsx:61]: I need to make the `/t` optional however. As the page needs to work with user projects as well.

**Comment by AlexTugarev** [comment]
Hopefully, I did test all cases. It worked nicely 💯 

/lgtm

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: be338cf2c06ea4a5a63f7a1dcb3624d9a01b1526</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5544#issuecomment-914156931" title="LGTM">AlexTugarev</a>*

Associated issue: *#5465*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [AlexTugarev]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 1aa844010e1e27fd1678ea3d269993c0f41ec9c6</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5544#issuecomment-914184104" title="LGTM">AlexTugarev</a>*

Associated issue: *#5465*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [AlexTugarev]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**